### PR TITLE
cartesian to cylindrical coordinate transformation

### PR DESCRIPTION
**Pull request by hongkai-dai**
This transformation is going to be used in the compliant manipulation for the valve task. It transform a point or a body orientation from the cartesian coordinate to a point / orientation in the cylindrical coordinate.

@tkoolen, could you review it?

### CODE REVIEW

File: solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m [hunk 1]
```diff
@@ -0,0 +1,49 @@
+classdef Cartesian2CylindricalTransform < drakeFunction.DrakeFunction
+  % Given a cartesian coordinate frame, transform its coordinates to a
+  % cylindrical coordinates
+  properties
+    T_cylinder % This is a homogeneous transformation matrix, T_cylinder transforms a cylinder with [0;0;1] being its axis, and [1;0;0] as its theta=0 radius, to the cylinder in the cartesian frame
```
Comment (tkoolen) [inline solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m]: The part 'and [1;0;0] as its theta=0 radius' is a little hard to understand. Maybe change it to 'and the angle theta measured with respect to the x-axis'.

File: solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      end
+      obj.T_cylinder = T_cylinder;
+    end
+    
+    function [x_cylinder,J] = eval(obj,x_cartesian)
+      % x_cylinder = [radius;theta;height;roll;pitch;yaw]
+      % we attach a cartesian frame [x_tangent;y_normal;z_tangent] in R^3x3 at the tip of the
+      % radius, where x_tangent is aligned with cross(height,radius),
+      % y_norm is along the radius direction, and z_tangent =
+      % cross(x_tangent,y_normal). roll;pitch;yaw are the Euler angles that
```
Comment (tkoolen) [inline solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m]: so is z_tangent always equal to the height axis of the cylinder?

File: solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      if(~isHT(T_cylinder))
+        error('T_cylinder should be a homogenous transformation matrix');
+      end
+      obj.T_cylinder = T_cylinder;
+    end
+    
+    function [x_cylinder,J] = eval(obj,x_cartesian)
+      % x_cylinder = [radius;theta;height;roll;pitch;yaw]
+      % we attach a cartesian frame [x_tangent;y_normal;z_tangent] in R^3x3 at the tip of the
+      % radius, where x_tangent is aligned with cross(height,radius),
```
Comment (tkoolen) [inline solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m]: should be cross(radius, height)

File: solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m (hunk 2 above)
Comment (tkoolen) [inline solvers/+drakeFunction/+geometry/Cartesian2CylindricalTransform.m]: investigated this, and yes it matches as long as x_tangent is indeed cross(radius, height)

**Comment by tkoolen** [comment]
Looks good, but why pass in rpy and not a quaternion? Also, is the contract of a drakeFunction that the Nth output argument is the gradient of the N-1th? Because if that's not the contract, in my opinion it would be better to have the rotation part of J be a mapping from angular velocity in the cartesian frame to angular velocity in the 'rotation frame' associated with the cylindrical coordinate system.

**Comment by hongkai-dai** [comment]
To the quaternion question, in the manipulation task, I think the Euler
angles are more meaningful than the quaternions. We might want to tolerate
a rotation error around the x_tangent axis, but not around the z_tangent
axis.

The contract in drakeFuntion is that the 2nd output is the gradient of the
first output, we can implement a separate function to find the mapping from
the angular velocity in the cartesian space to the angular velocity of the
rotation frame.

File: solvers/+drakeFunction/+geometry/doc_cartesian2cylindrical.tex [hunk 4]
```diff
@@ -0,0 +1,17 @@
+\documentclass{article}
```
Comment (RussTedrake) [inline solvers/+drakeFunction/+geometry/doc_cartesian2cylindrical.tex]: can we put these docs in the drake/doc directory instead?  (and update the reference to them in the comments).

**Comment by RussTedrake** [comment]
otherwise looks good.  ready to accept.

**Comment by hongkai-dai** [comment]
I have moved the documentation to the doc directory

**Comment by hongkai-dai** [comment]
I just made a new change to this pull request, now the cylindrical coordinate is constructed by given the cylinder axis, the cylinder x direction and cylinder origin in the Cartesian coordinate. This is easier to understand than using a homogeneous transform

**Pull request merged**